tests/js/common/shell/shell-one-shard.js

```diff
@@ -81,6 +102,36 @@ function OneShardPropertiesSuite () {
         assertEqual(props.replicationFactor, undefined);
       }
     },
+
+    testDeviatingWriteConcernAndMinReplicationFactorForDatabase : function () {
+      if (!isCluster) {
+        return;
+      }
+      try {
+        db._createDatabase(dn, { replicationFactor: 2, minReplicationFactor: 1, writeConcern: 2, sharding: "flexible" });
+        fail();
+      } catch (err) {
+        assertEqual(ERRORS.ERROR_BAD_PARAMETER.code, err.errorNum);
+      }
+        
+      assertTrue(db._createDatabase(dn, { replicationFactor: 2, minReplicationFactor: 2, writeConcern: 2, sharding: "flexible" }));
+    },
+    
+    testDeviatingWriteConcernAndMinReplicationFactorForCollection : function () {
```

Comment:
Could we also add this database level based test also for collection level as well?